src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MicrosoftQualityGuidelinesAnalyzersResources.resx

```diff
@@ -252,4 +252,13 @@
   <data name="AvoidPropertySelfAssignmentMessage" xml:space="preserve">
     <value>The property {0} should not be assigned to itself.</value>
   </data>
+  <data name="ReferencingAndReassigningObjectInSameStatementMessage" xml:space="preserve">
+    <value>Object '{0}' is referenced and reassigned in the same statement. You are at risk of referring to unintended object.</value>
+  </data>
+  <data name="ReferencingAndReassigningObjectInSameStatementTitle" xml:space="preserve">
+    <value>Referring to object and reassigning it in the same statement.</value>
+  </data>
+  <data name="ReferencingAndReassigningObjectInSameStatementDescription" xml:space="preserve">
+    <value>When you go through the assignment chain right to left, firstly, reference in object '{0}' will be changed, but when you reffer to this object second time, since it's still the same statement, it will be pointing to it's old value. So if old value of '{0}' isn't saved elsewhere, result of the latter assignment will be lost.</value>
```

Comment:
```suggestion
    <value>Referencing and reassigning an object in the same statement is not recommended. After the value of the object is changed, its value is referenced. Since it's in the same statement, the referenced value is the previous value before the reassignment. Accordingly, the reassignment is lost and the object retains its previous value.</value>
```